packages/store-ui/src/ProductQuantity/NumericStepper.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { FC, Fragment, useCallback } from 'react'
+import { Button, Flex, Input } from 'theme-ui'
+
+export interface NumericStepperProps {
+  variant?: string
+  minValue: number | 1
+  maxValue?: number | undefined
+  onChange: (quantity: number) => void
+}
+
+export const NumericStepper: FC<NumericStepperProps> = ({
+  variant = 'productQuantity',
+  minValue,
+  maxValue,
+  onChange,
+  children,
+}) => {
+  var [quantity, setQuantity] = React.useState<number>(1)
+
+  const updateQuantity = useCallback(
+    (value: number) => {
```

Comment:
you can apply the maxValue/minValue just here, doing something like:

```tsx
const normalizedValue = Math.min(Math.max(value, minValue), maxValue)
```
 to do that, you just need to put in the default value for `maxValue = Infinity` in the destructuring arguments

and you can validate for doesn't call `setQuantity/onChange` when the value remains equals to current `quantity`

with this validations here you can just use `onClick={() => updateQuantity(quantity - 1)}` in the Button

Reply:
thank you! This does make it much better!